packages/lodestar/src/chain/clock/local/LocalClock.ts

```diff
@@ -0,0 +1,80 @@
+import {IBeaconClock} from "../interface";
+import {IBeaconConfig} from "@chainsafe/eth2.0-config";
+import {computeEpochAtSlot, getCurrentSlot} from "@chainsafe/eth2.0-state-transition";
+import {Epoch, Slot} from "@chainsafe/eth2.0-types";
+
+type NewSlotCallback = (slot: Slot) => void;
+type NewEpochCallback = (epoch: Epoch) => void;
+
+export class LocalClock implements IBeaconClock {
+
+  private readonly config: IBeaconConfig;
+  private readonly genesisTime: number;
+  private currentSlot: number;
+  private isRunning: boolean;
+  private newSlotCallbacks: NewSlotCallback[] = [];
+  private newEpochCallbacks: NewEpochCallback[] = [];
+
+
+  public constructor(config: IBeaconConfig, genesisTime: number) {
+    this.config = config;
+    this.genesisTime = genesisTime;
+    //this assumes clock time is trusted
+    this.currentSlot = getCurrentSlot(this.config, this.genesisTime);
+  }
+
+  public async start(): Promise<void> {
+    this.isRunning = true;
+    const diffTillNextSlot = this.getDiffTillNextSlot();
+    setTimeout(
```

Comment:
Can we keep track of the timerId here and below, so we can preemptively cancel on `stop`? small race condition where the timer will be called after stop.